public/app/types/alerting.ts

```diff
@@ -61,31 +61,43 @@ export interface NotifierDTO {
   optionsTemplate: string;
   type: NotifierType;
   heading: string;
-  options: Option[];
+  options: NotificationChannelOption[];
   info?: string;
+  secure?: boolean;
 }
 
-export interface NotificationChannel {
+export interface NotificationChannelType {
   value: string;
   label: string;
   description: string;
   type: NotifierType;
   heading: string;
-  options: Option[];
+  options: NotificationChannelOption[];
   info?: string;
 }
 
 export interface NotificationChannelDTO {
-  [key: string]: string | boolean | SelectableValue<string>;
+  [key: string]: string | boolean | number | SelectableValue<string>;
+  id: number;
   name: string;
   type: SelectableValue<string>;
   sendReminder: boolean;
   disableResolveMessage: boolean;
   frequency: string;
   settings: ChannelTypeSettings;
+  secureSettings: NotificationChannelSecureSettings;
+  secureFields: NotificationChannelSecureFields;
   isDefault: boolean;
 }
 
+export interface NotificationChannelSecureSettings {
+  [key: string]: string | number;
+}
+
+export interface NotificationChannelSecureFields {
+  [key: string]: boolean;
+}
+
```

Comment:
These could be defined as Records:

```suggestion
export type NotificationChannelSecureSettings = Record<string, string | number>
export type NotificationChannelSecureFields = Record<string, boolean>
```